### ISSUE TITLE
lib's libavformat.a

### ISSUE THREAD

**Issue by lwalmeida** [jacobbo/WebEye#21]
Hello, I have been studying your project and to compile the project and c ++ "StreamPlayer" gives an error in lib's libavformat.a, libavcodec.a and in the others.
From where do you generate these lib's to compile the project?

Thanks
Luis Wagner de Almeida

**Comment by jacobbo** [comment]
Hi Luis Wagner,

These files are the FFmpeg library and you need to build them first.
You can find instructions on how to do it here:
https://trac.ffmpeg.org/wiki/CompilationGuide/MSVC

Regards,
Alexander

**Comment by tasoss** [comment]
Hello.
I have managed to compile StreamPlayer using the latest ffmpeg version.
The dll produced is StreamPlayer.dll.
I added it as a resource to the project but LoadLibrary fails.
My produced dll is 1.5MB but yours is 11MB.
Could you please help me?
BTW thanks for the great job!

**Comment by jacobbo** [comment]
Hi there, I used the following options to build FFmpeg:

	// The FFmpeg framework built using the following configuration:
	// x64: ./configure --toolchain=msvc --arch=amd64 --target-os=win64 --enable-version3 --enable-static --disable-shared --disable-programs --disable-doc
    // x86: ./configure --toolchain=msvc --arch=i386 --enable-version3 --enable-static --disable-shared --disable-programs --disable-doc

**Comment by tasoss** [comment]
Very quick reply :)
Well my problem is not FFmpeg compilation.
It's the dll that is produced by streamplayer (streamplayer.dll)
which can't be loaded via LoadLibrary(called in StreamPlayerProxy.cs).
Any ideas? :)

**Comment by jacobbo** [comment]
There are two dlls: streamplayerx86.dll & streamplayerx64.dll. They both are quite big because they link FFmpeg statically. The StreamPlayerControl then loads an appropriate version of the dll (x86 or x64) depending on platform it's running on.

**Comment by tasoss** [comment]
Ok will recompile everything again and let you know.
Thanks again!
ps:They(ffmpeg) have deprecated some methods that you are using.
I will try to correct a few of them(as much as my c++ knowledge can help) and make a pull request.

**Comment by jacobbo** [comment]
You can suppress this deprecation stuff using compiler options, as far as I remember.

**Comment by tasoss** [comment]
Indeed but it would be nice to use their new stuff.
BTW are you still working on the control?

**Comment by jacobbo** [comment]
Not really. My last idea was to use the sources from ffplay.exe (with minimal changes) to render video & audio & subtitles, but that was like a year ago :)

**Comment by tasoss** [comment]
That's a shame because it's working very very nicely!
I would like to see if it's easy to add hardware support.
But first things first.Let's try to create the correct dll :)

**Comment by tasoss** [comment]
I think the part that i'm missing is to statically add ffmpeg to streamplayer.dll.
Any advice? :)

**Comment by jacobbo** [comment]
The procedure is the following:
1. Build FFmpeg as a static library
2. Reference it in the C++ project (StreamPlayer.vcxproj)
3. Build the StreamPlayer.vcxproj project and get StreamPlayer.dll
4. Add StreamPlayer.dll to the C# project (StreamPlayerControl.csproj) as a resource
5. Repeat steps 1-4 both for x86 and x64 configs.

You can use the configs from my answer above to build FFmpeg statically both x86 & x64

**Comment by tasoss** [comment]
Thank you very much you are very kind!!
I will try it and let you know.
Thanks again!
